hail/src/main/scala/is/hail/expr/ir/TableIR.scala

```diff
@@ -1466,7 +1466,7 @@ case class CastMatrixToTable(
   colsFieldName: String
 ) extends TableIR {
 
-  def typ: TableType = LowerMatrixIR.loweredType(child.typ, entriesFieldName, colsFieldName)
+  lazy val typ: TableType = LowerMatrixIR.loweredType(child.typ, entriesFieldName, colsFieldName)
```

Comment:
Oh dear. This is basically all the IR nodes that the combiner uses other than Map. :scream: